keep/providers/mongodb_provider/mongodb_provider.py

```diff
@@ -0,0 +1,126 @@
+import os
+from pymongo import MongoClient
+from keep.contextmanager.contextmanager import ContextManager
+from keep.providers.base.base_provider import BaseProvider
+from keep.providers.models.provider_config import ProviderConfig, ProviderScope
+import pydantic
+import dataclasses
+
+@pydantic.dataclasses.dataclass
+class MongoDBProviderAuthConfig:
+    uri: str | None = dataclasses.field(
+        metadata={"required": False, "description": "MongoDB connection URI"}
+    )
+    username: str = dataclasses.field(
+        metadata={"required": False, "description": "MongoDB username"}
+    )
+    password: str = dataclasses.field(
+        metadata={"required": False, "description": "MongoDB password", "sensitive": True}
+    )
+    host: str = dataclasses.field(
+        metadata={"required": False, "description": "MongoDB hostname"}
+    )
+    database: str = dataclasses.field(
+        metadata={"required": False, "description": "MongoDB database name"}
+    )
+
+class MongoDBProvider(BaseProvider):
+    """Enrich alerts with data from MongoDB."""
+
+    PROVIDER_SCOPES = [
+        ProviderScope(
+            name="connect_to_server",
+            description="The user can connect to the server",
+            mandatory=True,
+            alias="Connect to the server",
+        )
+    ]
+
+    def __init__(
+        self, context_manager: ContextManager, provider_id: str, config: ProviderConfig
+    ):
+        super().__init__(context_manager, provider_id, config)
+        self.client = None
+
+    def validate_scopes(self):
+        """
+        Validates that the user has the required scopes to use the provider.
+        """
+        try:
+            client = self.__generate_client()
+            client.close()
+            scopes = {
+                "connect_to_server": True,
+            }
+        except Exception as e:
+            self.logger.exception("Error validating scopes")
+            scopes = {
+                "connect_to_server": str(e),
+            }
+        return scopes
+
+    def __generate_client(self):
+        """
+        Generates a MongoDB client.
+
+        Returns:
+            pymongo.MongoClient: MongoDB Client
+        """
+        if self.authentication_config.uri:
+            client = MongoClient(self.authentication_config.uri)
+        else:
+            client = MongoClient(
+                f"mongodb://{self.authentication_config.username}:{self.authentication_config.password}@{self.authentication_config.host}/{self.authentication_config.database}"
+            )
```

Comment:
What if the config values are empty (considering they are not required), can we handle the error case here?